### ISSUE TITLE
Unexpected keyword argument 'sysroot' in SLES 11 SP4

### ISSUE THREAD

**Issue by NesManrique** [sosreport/sos#877]
Hello,

I just cloned sosreport into a SLES 11 SP4 installation, using python 2.6, and tried running it right away but this error came out. 

sos-master/sosreport -a
Traceback (most recent call last):
  File "sos-master/sosreport", line 25, in <module>
    main(sys.argv[1:])
  File "/root/sos-master/sos/sosreport.py", line 1636, in main
    sos = SoSReport(args)
  File "/root/sos-master/sos/sosreport.py", line 697, in **init**
    self.policy = sos.policies.load(sysroot=self.opts.sysroot)
  File "/root/sos-master/sos/policies/**init**.py", line 38, in load
    cache['policy'] = policy(sysroot=sysroot)
TypeError: **init**() got an unexpected keyword argument 'sysroot

I also tried upgrading python to the latest 2.7 version and installing sosreport using pip but I get the exact same error. I'm aware there is an [old issue related to this error](https://github.com/sosreport/sos/issues/809) but don't know why it is happening again.

Regards,

Nestor

**Comment by bmr-cymru** [comment]
On Fri, Sep 16, 2016 at 12:46:00PM -0700, Nestor Manrique wrote:
> sos-master/sosreport -a
> Traceback (most recent call last):
>   File "sos-master/sosreport", line 25, in <module>
>     main(sys.argv[1:])
>   File "/root/sos-master/sos/sosreport.py", line 1636, in main
>     sos = SoSReport(args)
>   File "/root/sos-master/sos/sosreport.py", line 697, in __init__
>     self.policy = sos.policies.load(sysroot=self.opts.sysroot)
>   File "/root/sos-master/sos/policies/__init__.py", line 38, in load
>     cache['policy'] = policy(sysroot=sysroot)
> TypeError: __init__() got an unexpected keyword argument 'sysroot

Hi Nestor,

Do you have anything set for PYTHONPATH, PYTHONHOME, or other py
environment variables?

If you've cloned current master (and the line numbers in your
backtrace look correct), then you should have this in the file
sos/policies/suse.py:

 26 class SuSEPolicy(LinuxPolicy):
 27     distro = "SuSE"
 28     vendor = "SuSE"
 29     vendor_url = "https://www.suse.com/"
 30     _tmp_dir = "/var/tmp"
 31 
 32     def __init__(self, sysroot=None):
 33         super(SuSEPolicy, self).__init__()

Which would make the error you're seeing impossible (since no
other policy class should return True for the policy.check()
class method).

You can also check that there are no stray files in the
sos/policies directory by checking that everything has the
sysroot parameter:

$ grep '__init__.*sysroot=' sos/policies/*.py

You should see matches for debian, ibmkvm, __init__, redhat,
suse, and ubuntu.

Regards,
Bryn.

**Comment by NesManrique** [comment]
Hello Bryn,

I think I found the issue. There was indeed some sysroot parameters missing in the OpenSuSEPolicy init method, lines 107 and 108 of sos/policies/suse.py.

```
90 class OpenSuSEPolicy(SuSEPolicy):
91    distro = "OpenSuSE"
92    vendor = "SuSE"
93    vendor_url = "https://www.opensuse.org/"
    .
    .
    .
107    def __init__(self):
108        super(OpenSuSEPolicy, self).__init__()
```

It should be like this

```
90 class OpenSuSEPolicy(SuSEPolicy):
91    distro = "OpenSuSE"
92    vendor = "SuSE"
93    vendor_url = "https://www.opensuse.org/"
    .
    .
    .
107    def __init__(self,sysroot=None):
108        super(OpenSuSEPolicy, self).__init__(sysroot=sysroot)
```

Thank you for your help!

Nestor

**Comment by bmr-cymru** [comment]
Hi Nestor,

Definitely an sos bug then - we'll get it fixed in the next release.

Bryn.

**Comment by bmr-cymru** [comment]
Fixed last year, in commit 472c8193a9f69bb7bd352203ef3404b0400a0d4a:

```
commit 472c8193a9f69bb7bd352203ef3404b0400a0d4a
Author: Bryn M. Reeves <bmr@redhat.com>
Date:   Thu May 5 11:47:22 2016 +0100

    [suse] fix constructor sysroot argument
```
